### PR TITLE
Changed mouse button to object

### LINKED ISSUE #6847: mouseButton contains incorrect value after pressing both mouse buttons and releasing one of them
### Most appropriate sub-area of p5.js?

- [ ] Accessibility
- [ ] Color
- [ ] Core/Environment/Rendering
- [ ] Data
- [ ] DOM
- [X] Events
- [ ] Image
- [ ] IO
- [ ] Math
- [ ] Typography
- [ ] Utilities
- [ ] WebGL
- [ ] Build process
- [ ] Unit testing
- [ ] Internationalization
- [ ] Friendly errors
- [ ] Other (specify if possible)

### p5.js version

1.9.0

### Web browser and version

122.0.6261.95 (Official Build) (64-bit) (cohort: Stable) 

### Operating system

Windows 10 Home Edition

### Steps:
1. Press and hold left mouse button
1. Press and hold right mouse button
1. Release left mouse button
   * Expected: console logs `left`
   * Actual: console logs `right`

### Snippet:

```js
function setup() {
  createCanvas(400, 400);
}

function mouseReleased() {
  console.log(mouseButton);
}

function draw() {
  background(220);
}
```

### Additional Context

I discovered this issue when implementing the game of Minesweeper.  In that game, pressing and releasing both mouse buttons at the same time is the way to uncover all unmarked neighbors of a cell, and I found that I could not detect this condition reliably using `mouseButton`.  I believe this is due to `mouseButton` being updated in `mousePressed` but not `mouseReleased`.

https://github.com/processing/p5.js/blob/main/src/events/mouse.js#L657
https://github.com/processing/p5.js/blob/main/src/events/mouse.js#L733

I have a pull request (with associated unit test) that appears to resolve the issue.   This bug was previously reported in #5733, but the submitter resolved that issue as they found a workaround.

### PR DESCRIPTION
**Pull request by diyaayay**
Resolves #6847

 Changes:
- Changed `mouseButton` from a string to object.
- `mouseButton` now intends to track the current state of mouse buttons, showing which buttons are pressed at any given moment.

This work is done over the PR #7378

#### PR Checklist

- [x] `npm run lint` passes
- [x] [Inline documentation] is included / updated
- [x] [Unit tests] are included / updated

[Inline documentation]: https://github.com/processing/p5.js/blob/main/contributor_docs/inline_documentation.md
[Unit tests]: https://github.com/processing/p5.js/tree/main/contributor_docs#unit-tests

### CODE REVIEW

**Comment by diyaayay** [comment]
I think it makes more sense for `mouseButton.left` to remain true as long as any touch action is present on the screen. Setting it to false in `onpointerup`, using something like this, might work well:
``` js
if (this._activeTouches.size === 0) {
  Object.keys(this.mouseButton).forEach((key) => (this.mouseButton[key] = false));
}
```

This approach means we treat all active touches on the screen as left touches, correct? In that case, right and center clicks would only be supported by a mouse if I'm not wrong. @davepagurek

**Comment by davepagurek** [comment]
> This approach means we treat all active touches on the screen as left touches, correct?

I think this is fine if it ends up being required. Although it looks like on MDN, [`PointerEvent` extends `MouseEvent`](https://developer.mozilla.org/en-US/docs/Web/API/PointerEvent), so would that mean that when we receive a pointer event, we can check `e.buttons`? If so, it feels like maybe we don't need to do a special case for touches at all.

If we keep track of the buttons for each active touch in a similar way to how you currently do the global `mouseButton`, then after any pointer event, we could update just the active pointer from the current event, and then update the global `mouseButton` to match:

```js
const pointer = this.activePointers[e.pointerId]
pointer.mouseButton.left = (e.buttons & 1) !== 0;   
pointer.mouseButton.center = (e.buttons & 4) !== 0; 
pointer.mouseButton.right = (e.buttons & 2) !== 0;

for (const key in this.mouseButton) {
  this.mouseButton[key] = Object.values(this.activePointers).some((p) => p.mouseButton[key])
}
```

**Comment by diyaayay** [comment]
@davepagurek This now uses `_activePointers` to keep track of `mouseButton` by using the `buttons` property of each active pointer on the screen i.e. in the [Mouse-Button sketch](https://editor.p5js.org/diya.solanki.31/sketches/3P-fuscER) here, the `mouseButton.left` stays `true` for as long as there is a finger/touch on the screen. I've tested all the other sketches from the [Event-handling](https://github.com/processing/p5.js/pull/7378) PR ensure that nothing breaks there from these changes.

File: src/events/pointer.js [hunk 1]
```diff
@@ -691,24 +691,29 @@ function pointer(p5, fn){
...
+   * @property {boolean} mouseButton.right - Whether the right mouse button is pressed.
+   * @property {boolean} mouseButton.center - Whether the middle mouse button is pressed.
    * @readOnly
    *
    * @example
    * <div>
    * <code>
    * function setup() {
-   *   createCanvas(100, 100);
+   *   createCanvas(400, 400);
```
Comment (davepagurek) [inline src/events/pointer.js:716]: This might be a little large on the reference page, would the example still work OK with a smaller canvas like 200x200?

**Review by davepagurek — COMMENTED** [review]
This looks great, just one minor comment about the reference!

File: src/events/pointer.js (hunk 1 above)
Comment (diyaayay) [reply src/events/pointer.js:716]: This is what 200x200 looks like:
image
300x300 looks just fine:
image
We can also print all buttons separately to fit the 200x200 canvas size:
image
```js
p.text(`Left: ${p.mouseButton.left}`, p.width / 2, p.height / 2 - 20);
        p.text(`Right: ${p.mouseButton.right}`, p.width / 2, p.height / 2);
        p.text(`Center: ${p.mouseButton.center}`, p.width / 2, p.height / 2 + 20);```

Comment (davepagurek) [reply src/events/pointer.js:716]: The last one is fairly easy to read, let's go with that one. Thanks!

**Comment by limzykenneth** [comment]
@diyaayay @davepagurek Can we merge this now?

**Comment by davepagurek** [comment]
I think we're good to go!

**Pull request merged**